### ISSUE TITLE
permissions on dynaforms

### ISSUE THREAD

**Issue by kbower** [kbower/tickettest1#188]
_Imported from trac ticket 188.
Created by **rneville**
Opened in trac: May 13, 2010, 1:42:41 PM
Last modified in trac: May 21, 2010, 8:42:47 AM_
***
Permissions need to be enforced on dynaforms

**Comment by kbower** [comment]
_**rneville** trac comment on May 13, 2010, 2:05:35 PM:_

accepted

**Comment by kbower** [comment]
_**rneville** trac comment on May 21, 2010, 8:42:47 AM:_

(In [[523]](../commit/75acfaa138c01be7a7168d8a00f1d5138347d5b9)) fixes #188

## New Security Settings and Data Takeon
Initial groups and users moved from mockdata into dataload script. Data that is required for system function should be here as opposed to data used for testing and development (mockdata.sql).

System permissions have now been added in earnest to the dataload.py script. Whenever you add a new screen, you should, at a minimum add a new permission to control access to your screen to the list in dataload.py file. The format for doing so looks like this:

```
adminzone = Permission()
adminzone.permission_name = 'adminzone'
adminzone.description = 'ADMIN - Ability to create/edit delivery zones.'
DBSession.add(adminzone)
all_permissions.append(adminzone)
```
The permission name will be the string that you use in Flex or in Predicates in python to check for proper access. The description is what will show up in the admin screen.  Please follow the convention of all caps ADMIN, ORDERS, PRODUCTS, etc. as the first part, a space, a dash and another space, then the description. This way it will be easy to sort permissions based on the section of the application to which they apply. Also, please pay attention to the fact the permission definitions in dataload.py are organized the same way...please add your permissions to the appropriate area in that file.

The XML data for the dynamic screens in ADMIN now have a permission attribute...this should match a permission_name as outlined above. At the screen level, you can control query, edit and delete discretely or you can use the same permission for all three. You can also choose to use a totally different permission name to control access to the screen altogether as well. Forms and menus for which you do not have access will appear greyed out and you will get an error now if you try and access them without the proper permissions.

**Important: Not the usual yadda-yadda**
If you are adding "admin-ny" type tables & services, you should be adding metadata to your actionscript TransferObject definition (see others for examples) and an entry in the XML menu in AdminView.mxml to create a dynamic maintenance screen for it. The metadata will control how that screen looks and behaves. The meta-markup language for this is still in flux and not well documented...sorry...like I said, look at other examples. If you have a new situation that you think should be covered, open a ticket and assign it to me and I'll look at it. You should also be adding the minimum set of permissions.

**Comment by kbower** [comment]
_**rneville** trac status change on May 21, 2010, 8:42:47 AM:_ **closed**
